src/storage/compaction_test/src/server.rs

```diff
@@ -185,18 +195,19 @@ pub async fn compaction_test_serve(
                 new_committed_epoch
             );
             assert_eq!(max_committed_epoch, new_committed_epoch);
-            tracing::info!(
-                "Check results for version: id: {}, epochs: {:?}",
-                new_version_id,
-                epochs,
-            );
 
             local_version_manager.try_update_pinned_version(
                 pin_version_response::Payload::PinnedVersion(new_version),
             );
 
-            check_compaction_results(&expect_results, &hummock, opts.table_id).await?;
+            let new_version_iters = open_hummock_iters(&hummock, &epochs, opts.table_id).await?;
 
+            // spawn a task to check the results
+            tokio::spawn(check_compaction_results(
```

Comment:
Should we join these task to ensure they are finished?

Reply:
Good catch.